app/Http/Controllers/Api/ApiPetController.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Pet;
+use Validator;
+use App\Contact;
+use Illuminate\Http\Request;
+use Illuminate\Database\QueryException;
+use App\Http\Resources\Pet\Pet as PetResource;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+
+class ApiPetController extends ApiController
+{
+    /**
+     * Get the detail of a given pet.
+     * @param  Request $request
+     * @return \Illuminate\Http\Response
+     */
+    public function show(Request $request, $id)
+    {
+        try {
+            $pet = Pet::where('account_id', auth()->user()->account_id)
+                ->where('id', $id)
+                ->firstOrFail();
+        } catch (ModelNotFoundException $e) {
+            return $this->respondNotFound();
+        }
+
+        return new PetResource($pet);
+    }
+
+    /**
+     * Store the pet.
+     * @param  Request $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        // Validates basic fields to create the entry
+        $validator = Validator::make($request->all(), [
+            'pet_category_id' => 'integer|required|exists:pet_categories,id',
+            'contact_id' => 'required|integer',
+        ]);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "contact_id" 7 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1192)